autonomous/autonomous.py

```diff
@@ -64,10 +111,83 @@ def drive_to_shoot(self, initial_call) -> None:
+        bottomLeft = Translation2d(self.goal.X() - 0.5, self.goal.Y() - 0.5)
+        traj_config.addConstraint(
+            RectangularRegionConstraint(
+                bottomLeft, topRight, MaxVelocityConstraint(0.5)
+            )
+        )
+
+        """Generates a trajectory to self.goal and displays it"""
+        x_controller = PIDController(2.5, 0, 0)
+        y_controller = PIDController(2.5, 0, 0)
+        heading_controller = ProfiledPIDControllerRadians(
+            3, 0, 0, TrapezoidProfileRadians.Constraints(2, 2)
+        )
+        heading_controller.enableContinuousInput(math.pi, -math.pi)
+
+        self.drive_controller = HolonomicDriveController(
+            x_controller, y_controller, heading_controller
+        )
+
+        chassis_velocity = self.chassis.get_velocity()
+        chassis_speed = math.hypot(chassis_velocity.vx, chassis_velocity.vy)
+        pose = self.chassis.get_pose()
+
+        next_pos = waypoints[0] if waypoints else self.goal.translation()
+        translation = next_pos - pose.translation()
+
+        # Generating a trajectory when the robot is very close to the goal is unnecesary, so this
+        # return an empty trajectory that starts at the end point so the robot won't move.
+        distance_to_goal = (self.goal.translation() - pose.translation()).norm()
+        if distance_to_goal <= 0.01:
+            return Trajectory([Trajectory.State(0, 0, 0, pose)])
+
+        # Since robot is stationary from one action to another, point the control vector at the goal to avoid the robot taking unnecessary turns before moving towards the goal
+        kD = 0.3
+
+        spline_start_momentum_x = translation.x * kD
+        spline_start_momentum_y = translation.y * kD
+
+        goal_spline = Spline3.ControlVector(
+            (self.goal.X(), self.goal.rotation().cos() * self.END_CONTROL_SCALER),
+            (self.goal.Y(), self.goal.rotation().sin() * self.END_CONTROL_SCALER),
+        )
+
+        start_point_spline = Spline3.ControlVector(
+            (pose.x, spline_start_momentum_x),
+            (pose.y, spline_start_momentum_y),
+        )
+
+        traj_config.setStartVelocity(chassis_speed)
+        try:
+            trajectory = TrajectoryGenerator.generateTrajectory(
+                start_point_spline, waypoints, goal_spline, traj_config
+            )
+        except RuntimeError:
+            return Trajectory([Trajectory.State(0, 0, 0, pose)])
+
+        self.robot_object = self.field.getObject("auto_trajectory")
+        self.robot_object.setTrajectory(trajectory)
+        return trajectory
+
+    def is_at_goal(self) -> bool:
+        real_at_goal = (
+            self.goal.translation() - self.chassis.get_pose().translation()
+        ).norm() < self.POSITION_TOLERANCE and abs(
+            (self.goal.rotation() - self.chassis.get_rotation()).radians()
+        ) < self.ANGLE_TOLERANCE
+        return real_at_goal
```

Comment:
Can you just return that statement or change the name? I dont think ```real_at_goal``` makes sense